src/main/sensors/gyro.c

```diff
@@ -160,18 +171,34 @@ void gyroUpdate(void)
 
     alignSensors(gyroADC, gyroADC, gyroAlign);
 
-    if (gyroConfig()->soft_gyro_lpf_hz) {
-        if (!gyroFilterStateIsSet) {
-            initGyroFilterCoefficients();
-        }
-        for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
-            gyroADC[axis] = lrintf(applyBiQuadFilter((float)gyroADC[axis], &gyroFilterState[axis]));
-        }
-    }
-
     if (!isGyroCalibrationComplete()) {
         performAcclerationCalibration(gyroConfig()->gyroMovementCalibrationThreshold);
     }
 
     applyGyroZero();
+
+    if (gyroConfig()->gyro_soft_lpf_hz) {
+        for (int axis = 0; axis < XYZ_AXIS_COUNT; axis++) {
+
+            if (debugMode == DEBUG_GYRO)
+                debug[axis] = gyroADC[axis];
+
+            if (gyroConfig()->gyro_soft_type == FILTER_BIQUAD)
```

Comment:
I chose to leave pt1 in there as it is easy in use and also much easier to apply to variable deltas in case it would be needed ever.